noir-projects/aztec-nr/aztec/src/oracle/storage.nr

```diff
@@ -1,42 +1,58 @@
-use dep::protocol_types::traits::Deserialize;
+use dep::protocol_types::{address::AztecAddress, traits::Deserialize};
 
 #[oracle(storageRead)]
-unconstrained fn storage_read_oracle<N>(storage_slot: Field, length: Field) -> [Field; N] {}
-
-unconstrained pub fn raw_storage_read<N>(storage_slot: Field) -> [Field; N] {
-    storage_read_oracle(storage_slot, N)
+unconstrained fn storage_read_oracle<N>(
+    storage_slot: Field,
+    length: Field,
+    address: Field,
+    block_number: Field
+) -> [Field; N] {}
+
+unconstrained pub fn raw_storage_read<N>(
+    storage_slot: Field,
+    address: AztecAddress,
+    block_number: u32
+) -> [Field; N] {
+    storage_read_oracle(storage_slot, N, address.to_field(), block_number as Field)
 }
 
-unconstrained pub fn storage_read<T, N>(storage_slot: Field) -> T where T: Deserialize<N> {
-    T::deserialize(raw_storage_read(storage_slot))
+unconstrained pub fn storage_read<T, N>(
```

Comment:
Find the ordering of arguments slightly weird here as it is different to what you are using later. The types should throw, but still a bit annoying in my mind.

Reply:
Hm you're right, I did the noir side first without looking at the node interface. Will adjust these to make them match.